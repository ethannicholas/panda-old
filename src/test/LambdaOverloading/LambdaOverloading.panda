method doSomething(fn:(Int)=>(Int)) {
    Console.writeLine("I am doing something to 42: \{fn(42)}")
}

method doSomething(fn:(Int)=>(Real)) {
    Console.writeLine("I am doing something real to 42: \{fn(42)}")
}

method doSomething(fn:(Int)=>(String)) {
    Console.writeLine("I am doing something stringy to 42: \{fn(42)}")
}

doSomething(x => x * 2)
doSomething(x => x * 2.8)
doSomething(x => x + " is the answer!")