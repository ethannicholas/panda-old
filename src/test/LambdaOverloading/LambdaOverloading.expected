I am doing something to 42: 84
I am doing something real to 42: 117.6
I am doing something stringy to 42: 42 is the answer!
